test/integration_tests.jl

```diff
@@ -4,27 +4,43 @@
 # You should add the PAT as an encrypted Travis CI environment variable named `BCBI_TEST_USER_GITHUB_TOKEN`.
 # You should generate an SSH deploy key. Upload the public key as a deploy key for the testing repo.
 # You should make the private key an encrypted Travis CI environment variable named `COMPATHELPER_PRIV`. Probably you will want to Base64-encode it.
+const GLOBAL_PR_TITLE_PREFIX = Random.randstring(8)
+const GITHUB = "GITHUB"
+const GITLAB = "GITLAB"
 
-@testset "GitHub Integration Test" begin
-    GLOBAL_PR_TITLE_PREFIX = Random.randstring(8)
-    COMPATHELPER_INTEGRATION_TEST_REPO = ENV["COMPATHELPER_INTEGRATION_TEST_REPO"]
-    TEST_USER_GITHUB_TOKEN = ENV["BCBI_TEST_USER_GITHUB_TOKEN"]
+function get_api(service_name, personal_access_token)
+    if service_name == GITHUB
+        token = GitHub.Token(personal_access_token)
+        return GitHub.GitHubAPI(; token=token)
+    elseif service_name == GITLAB
+        token = GitLab.PersonalAccessToken(personal_access_token)
+        return GitLab.GitLabAPI(; token=token)
+    end
+end
 
-    api = GitHub.GitHubAPI(; token=GitHub.Token(TEST_USER_GITHUB_TOKEN))
-    user, _ = GitForge.get_user(api)
-    repo, _ = GitForge.get_repo(api, COMPATHELPER_INTEGRATION_TEST_REPO)
+function ci_config(service_name)
+    if service_name == GITHUB
+        return CompatHelper.GitHubActions()
+    elseif service_name == GITLAB
+        return CompatHelper.GitLabCI()
+    end
+end
 
-    url = "https://$(user.login):$(TEST_USER_GITHUB_TOKEN)@github.com/$(COMPATHELPER_INTEGRATION_TEST_REPO)"
+function get_url(service, user, personal_access_token, test_repo)
+    login = if service == GITHUB
+        user.login
+    elseif service == GITLAB
+        user.username
+    end
 
-    # Setup variables used in CompatHelper.main()
-    env = Dict(
-        "GITHUB_REPOSITORY" => COMPATHELPER_INTEGRATION_TEST_REPO,
-        "GITHUB_TOKEN" => TEST_USER_GITHUB_TOKEN,
-    )
-    ci_cfg = CompatHelper.GitHubActions(
-        user.login, "41898282+github-actions[bot]@users.noreply.github.com"
-    )
+    return "https://$(login):$(personal_access_token)@$(lowercase(service)).com/$(test_repo)"
+end
```

Comment:
This could potentially be replaced by calling the `get_url_with_auth` function https://github.com/JuliaRegistries/CompatHelper.jl/blob/v3-dev/src/utilities/new_versions.jl#L151-L155